File: backend/users/urls.py
from django.urls import path
from . import views
from .views import CookieTokenObtainPairView, CookieTokenRefreshView, LogoutView, delete_user

urlpatterns = [
	path("register/", views.register_user, name="register-user"),
	path("login/", CookieTokenObtainPairView.as_view(), name="cookie_token_obtain_pair"),
	path("logout/", LogoutView.as_view(), name="logout"),
	path("auth-check/", views.check_auth, name="check-auth"),
	path("delete-user/", delete_user, name="delete-user"),
	path("token/refresh/", CookieTokenRefreshView.as_view(), name="token_refresh"),
]

File: frontend/src/utils/urls.js
const API_BASE = "/api";

const URLS = {
	// Public routes
	LOGIN: `${API_BASE}/users/login/`,
	REGISTER: `${API_BASE}/users/register/`,

	// Auth routes
	LOGOUT: `${API_BASE}/users/logout/`,
	CHECK_AUTH: `${API_BASE}/users/check-auth/`,
	REFRESH: `${API_BASE}/users/token-refresh/`,
	DELETE_USER: `${API_BASE}/users/delete-user/`,
};

export default URLS;
